src/classes/Player.js

```diff
@@ -132,6 +124,44 @@ class Player {
     }
   }
 
+  async createStream() {
+    const url = queue[this.songEntry];
+    if (url.includes('youtube.com')) {
+      return this.createYoutubeStream()
+
+    } else if (url.includes('soundcloud.com') && !!this.soundcloudClientID) {
+      return await this.createSoundcloudStream();
+    }
+  }
+
+  createYoutubeStream() {
+    const stream = ytdl(queue[this.songEntry], {
+      quality: 'highestaudio',
+      highWaterMark: 1 << 25
+    });
+
+    stream.once(streamEvents.info, ({ title }) => {
+      this.song = title;
+      if (!this.updateDispatcherStatus()) {
+        this.updateSongPresence();
+      }
+    });
+
+    return stream
+  }
+
+  async createSoundcloudStream() {
+    const stream = await scdl.download(queue[this.songEntry], this.soundcloudClientID);
+    const info = await scdl.getInfo(url, this.soundcloudClientID);
+
+    this.song = info.title;
+    if (!this.updateDispatcherStatus()) {
+      this.updateSongPresence();
+    }
+
+    return stream
+  }
+
   updateDispatcherStatus() {
```

Comment:
And because of the changes with how the streams are created, please add the following in `updateDispatcherStatus()`:

```js
if (!this.dispatcher) {
  return null;
}
```

The issue is that the stream creation requires the dispatcher to exist at the same time but it was my bad that I never added this check before because previously this call would be made after the dispatcher and streams are created.